src/browser/components/MainPage.jsx

```diff
@@ -39,6 +39,7 @@ export default class MainPage extends React.Component {
 
     this.state = {
       key,
+      sessionsExpired: new Array(this.props.teams.length),
```

Comment:
I originally proposed an idea to simplify the interface between the desktop app and the webapp (https://pre-release.mattermost.com/core/pl/qpezbgj1bb898m1q4uk8158x3a), but this proved to be out of scope for the requested feature.

Perhaps something to return to in the future!